tests/cloud_answer_tests_runner.py

```diff
@@ -0,0 +1,390 @@
+                {'Actual': '/usr/tmp/tmp43la9b0w.png',
+                'Expected': '/usr/tmp/tmpbpaqbgi3.png',
+                'Difference': '/usr/tmp/tmp43la9b0w-failed-diff.png'}
+            )]
+
+        missing_answers : list of string
+        Collection of missing answer tests specifying full name of the test.
+        eg. ['yt.visualization.tests.test_line_plots:test_multi_line_plot']
+
+    """
+    missing_answers = []
+    failed_answers = []
+
+    tree = ET.parse(nose_xml)
+    testsuite = tree.getroot()
+
+    for testcase in testsuite:
+        for error in testcase:
+            test_name = testcase.attrib["classname"] + ":" \
+                        + testcase.attrib["name"]
+            if "No such file or directory" in error.attrib["message"]:
+                missing_answers.append(test_name)
+            elif "Items are not equal" in error.attrib["message"]:
+                img_path = extract_image_locations(error.attrib["message"])
+                if img_path:
+                    failed_answers.append((test_name, img_path))
+    return failed_answers, missing_answers
+
+if __name__ == "__main__":
+    """Run answer tests on cloud platforms like Travis, Appveyor
+
+    This script executes yt answer tests on cloud platform.
+    If the test fail due to difference in actual and expected images, 
+    this function uploads a html page having all the plots which got failed 
+    (if executed with `-f` command line argument).
+    In case, answer store does not has a golden answer and if executed with 
+    `-m` argument, it uploads missing answers zip file to yt's curldrop server.
+    
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument("-f", "--upload-failed-tests", action="store_true",
+                        help="Upload a comparison report of failed answer tests"
+                             " to yt's curldrop server.")
+    parser.add_argument("-m", "--upload-missing-answers", action="store_true",
+                        help="Upload tests' answers that are not found in "
+                             "answer-store.")
+    args = parser.parse_args()
+
+    # ANSI color codes
+    COLOR_BLUE = '\x1b[34;1m'
+    COLOR_CYAN = '\x1b[36;1m'
+    COLOR_RESET = '\x1b[0m'
+
+    ANSWER_STORE = ytcfg.get("yt", "test_storage_dir")
+    if not ANSWER_STORE or ANSWER_STORE == "/does/not/exist":
+        ANSWER_STORE = "answer-store"
+
+    NOSETEST_XML = "answer_testing_nosetests.xml"
+
+    # first parameter is program name
+    # https://github.com/python/cpython/blob/master/Lib/unittest/main.py#L99
+    test_argv = [os.path.basename(__file__), '--with-answer-testing',
+                 '--nologcapture', '-s', '-d', '-v', '--local',
+                 '--attr=answer_test', '--local-dir=%s' % ANSWER_STORE,
+                 '--with-xunit', '--xunit-file=%s' % NOSETEST_XML, 'yt']
+    result = nose.run(argv=test_argv, addplugins=[AnswerTesting()], exit=False)
```

Comment:
Can we remove the `--attr "!answer_test"` from the nose command above, and then have this script just run as an `after_failure` on travis and operate purely on the xml?